src/main/java/com/google/step/servlets/UserServlet.java

```diff
@@ -70,25 +70,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
 
-    String idString;
+    long id;
     try {
-      idString = request.getPathInfo().substring(1);
-    } catch (IndexOutOfBoundsException e) {
+      String idString = request.getPathInfo().substring(1);
+      id = Long.parseLong(idString);
+    } catch (IndexOutOfBoundsException | NumberFormatException e) {
       response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
       return;
     }
-    String userEmail = userService.getCurrentUser().getEmail();
-    String username = (String) request.getParameter("username");
-    String bio = (String) request.getParameter("bio");
-
-    User user = userManager.readOrCreateUserByEmail(userEmail);
 
-    if (!idString.equals(String.valueOf(user.id))) {
+    String userEmail = userService.getCurrentUser().getEmail();
+    User user = userManager.readUser(id);
```

Comment:
This should now be wrapped in a try-catch and `SC_BAD_REQUEST` should be returned if the user does not exist?